ateam_kenobi/src/path_planning/path_planner.cpp

```diff
@@ -0,0 +1,233 @@
+// Copyright 2021 A Team
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+
+#include "path_planner.hpp"
+#include <ranges>
+#include <algorithm>
+#include <vector>
+#include <ateam_common/robot_constants.hpp>
+#include <ateam_geometry/ateam_geometry.hpp>
+
+namespace ateam_kenobi::path_planning
+{
+
+PathPlanner::PathPlanner()
+{
+}
+
+PathPlanner::Path PathPlanner::getPath(
+  const Position & start, const Position & goal, const World & world,
+  const std::vector<ateam_geometry::AnyShape> & obstacles,
+  const PlannerOptions & options)
+{
+  const auto start_time = std::chrono::steady_clock::now();
+
+  std::vector<ateam_geometry::AnyShape> augmented_obstacles = obstacles;
+
+  addRobotsToObstacles(world, start, augmented_obstacles);
+
+  if (options.use_default_obstacles) {
+    addDefaultObstacles(world, augmented_obstacles);
+  }
+
+  if (options.avoid_ball) {
+    augmented_obstacles.push_back(ateam_geometry::makeCircle(world.ball.pos, 0.04267 / 2));
```

Comment:
Why this number

Reply:
That is the diameter of a golf ball in meters.